node/node.go

```diff
@@ -661,10 +664,21 @@ func NewNode(config *cfg.Config,
 	// If an address is provided, listen on the socket for a connection from an
 	// external signing process.
 	if config.PrivValidatorListenAddr != "" {
+		protocol, address := tmnet.ProtocolAndAddress(config.PrivValidatorListenAddr)
 		// FIXME: we should start services inside OnStart
-		privValidator, err = createAndStartPrivValidatorSocketClient(config.PrivValidatorListenAddr, genDoc.ChainID, logger)
-		if err != nil {
-			return nil, fmt.Errorf("error with private validator socket client: %w", err)
+		switch protocol {
+		case "grpc":
+			grpcMetrics := grpc_prometheus.NewClientMetrics()
```

Comment:
It's unclear to me how to get this into the metrics. Prometheus seems to be global but I'm  unsure how to get this in there.

Reply:
The documentation for that function says:

>NewClientMetrics returns a ClientMetrics object. Use a new instance of ClientMetrics when not using the default Prometheus metrics registry, for example when wanting to control which metrics are added to a registry as opposed to automatically adding metrics via init functions.

I.e. this creates a new metrics registry instead of using the default registry -- you'll want to use `grpc_prometheus.DefaultClientMetrics` instead.

Reply:
works now thank you.